pkg/nsx/services/common/services.go

```diff
@@ -28,7 +28,7 @@ type SubnetServiceProvider interface {
 	GetSubnetByKey(key string) (*model.VpcSubnet, error)
 	GetSubnetByPath(path string) (*model.VpcSubnet, error)
 	GetSubnetsByIndex(key, value string) []*model.VpcSubnet
-	CreateOrUpdateSubnet(obj client.Object, vpcInfo VPCResourceInfo, tags []model.Tag) (string, error)
+	CreateOrUpdateSubnet(obj client.Object, vpcInfo VPCResourceInfo, tags []model.Tag, deprecated bool) (string, error)
```

Comment:
It seems `deprecated` means using a legacy API to create a Subnet here. Use `useLegacyAPI` or a more appropriate variable name here？

Reply:
Use useLegacyAPI instead, thanks!